app/loginComponent.js

```diff
@@ -157,12 +159,40 @@ class LoginComponent extends React.Component {
     });
   }
 
-  onPressRegister = () => {
-    this.signup(this.state.email, this.state.password);
+  onPressRegister = async () => {
+    try {
+      await this.signup(this.state.email, this.state.password);
+    } catch (error) {
+      Alert.alert("Registration error", error.message);
+    }
```

Comment:
Moved the try-catch for `onPressRegister` and `onPressLogin` out of the actual `login` and `signup` functions so they'd throw their errors and we can catch later and analyze. This allows us to check the error code and automatically register if we try to log them in and they don't already exist in the db.